app/auth/views.py

```diff
@@ -56,9 +60,9 @@ def create_user():
         if not isinstance(username, str):
             return response_message(
                 'Type Error', 'username must all be string', 400)
-        if not re.match("^[a-zA-Z0-9_.-]+$", username):
+        if not re.match("^[a-zA-Z0-9_.]+$", username):
             return response_message(
-                'Space Error', 'Username should not have a whitespace, better user _',
+                'Space Error', 'Username should not have a whitespace',
```

Comment:
How about we strip the username before, before checking if it matches the regex. Whitespaces shouldn't be a burden on the user